frontend/src/pages/ActivityReport/index.js

```diff
@@ -203,27 +207,45 @@ function ActivityReport({
 
         // The report can be edited if its in draft OR needs_action state.
 
-        const canWriteReport = (isCollaborator || isAuthor)
-          && (report.calculatedStatus === REPORT_STATUSES.DRAFT
-            || report.calculatedStatus === REPORT_STATUSES.NEEDS_ACTION);
+        const isMatchingApprover = report.approvers.filter((a) => a.User && a.User.id === user.id);
+
+        const canWriteAsCollaboratorOrAuthor = (isCollaborator || isAuthor)
+        && (report.calculatedStatus === REPORT_STATUSES.DRAFT
+          || report.calculatedStatus === REPORT_STATUSES.NEEDS_ACTION);
```

Comment:
Is the check for report status needed? I see a similar check for 'approverHasMarkedReport'.

Reply:
```approverHasMarkedReport``` is checking _approval_ status, the other checks are for report status. 

Collaborators & authors can edit when the report status is draft or needs action, approvers can only edit in the submitted state. Not sure if that answers your question or if I'm missing a way to clean it up a little further

Reply:
Yea far as I know there are two statuses Submission and Calculated. Submission status is more or less static. Calculated status reflects all the approver statuses. Yea so I think you are correct.